apps/client/src/features/cuesheet/cuesheet-progress/CuesheetProgress.tsx

```diff
@@ -0,0 +1,24 @@
+import MultiPartProgressBar from '../../../common/components/multi-part-progress-bar/MultiPartProgressBar';
+import { useTimer } from '../../../common/hooks/useSocket';
+import useViewSettings from '../../../common/hooks-query/useViewSettings';
+
+import styles from "./CuesheetProgress.module.scss"
+
+export default function CuesheetProgress() {
+  const { data } = useViewSettings();
+  const timer = useTimer();
+  const totalTime = (timer.duration ?? 0) + (timer.addedTime ?? 0);
+
+  return (
+    <MultiPartProgressBar
+      now={timer.current}
+      complete={totalTime}
+      normalColor={data!.normalColor}
+      warning={data!.warningThreshold}
+      warningColor={data!.warningColor}
+      danger={data!.dangerThreshold}
+      dangerColor={data!.dangerColor}
+      className={styles.progressOverride}
```

Comment:
Are these null assertions a good idea?

Reply:
I am sure there are better ideas

I had left a question for you related to this

> After reviewing the changes, I wonder whether should fetch the view changes directly?

The reason why this is passed in is to avoid storming the caching layer with requests. The nature of this component is for it to re-render several times a second

Depending on the approach we go here, we could move the request to the component that needs it or just type the hook properly since the data cannot be undefined